wasm/src/test_explorer.rs

```diff
@@ -0,0 +1,71 @@
+use qsc::{
+    compile,
+    hir::{Attr, PatKind},
+    PackageType,
+};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+use crate::{
+    project_system::{into_qsc_args, ProgramConfig},
+    STORE_CORE_STD,
+};
+
+#[wasm_bindgen]
+pub fn collect_test_callables(config: ProgramConfig) -> Result<Vec<String>, String> {
+    let (source_map, capabilities, language_features, _store, _deps) =
+        into_qsc_args(config, None).map_err(super::compile_errors_into_qsharp_errors_json)?;
+
+    let package = STORE_CORE_STD.with(|(store, std)| {
+        let (unit, _) = compile::compile(
+            store,
+            &[(*std, None)],
+            source_map,
+            PackageType::Lib,
+            capabilities,
+            language_features,
+        );
+        unit.package
+    });
+
+    let items_with_test_attribute = package
+        .items
+        .iter()
+        .filter(|(_, item)| item.attrs.iter().any(|attr| *attr == Attr::Test));
+
+    let callables = items_with_test_attribute
+        .filter(|(_, item)| matches!(item.kind, qsc::hir::ItemKind::Callable(_)));
```

Comment:
I wonder if this could be a place to take advantage of the `qsc_hir::globals::iter_package` that we have. I only know it's there because I recently used it for another change. Not sure if it adds much or not, but it might save you the work below of finding the parent namespace as it already has the logic within the iterator to collect the namespace as an array of strings.

Reply:
oh, hmm... but then you'd end up having to do extra work to look up the items again to see if they have the test attribute or modify the `Global` to include something that has the information you need.

Reply:
@swernli  and I had a discussion about this and the end result was to add a pass that detected test attribute issues and returned more user-friendly errors. I've added this pass in 5a28ef104b5d9c0e23ec2d2c76c9c66d797f42a7